### PR TITLE
chore(compactor2): Improve documentation on split times

### PR DESCRIPTION
**Pull request by alamb**
# Rationale:
I misunderstood the meaning of split_times when implementing compactor2 tests. I have fixed the code in https://github.com/influxdata/influxdb_iox/pull/7098 but I wanted to ensure the docs were clearer

# Changes
Clarify how the split times are used in the reorg planner

### CODE REVIEW

File: iox_query/src/frontend/reorg.rs [hunk 1]
```diff
@@ -107,16 +109,31 @@ impl ReorgPlanner {
...
-    /// Stream i (0 < i < split_times's length): Rows that have  `time` in range `(split_times[i-1], split_times[i]]`
-    /// Stream n (n = split_times.len()): Rows that have `time` *after* all the split_times and NULL rows
     ///
-    /// For example, if the input looks like:
+    /// Stream i, where 0 < i < split_times.len(): Rows that have
+    /// `time` in range `(split_times[i-1], split_times[i]]` (greater than split_times[-1]
+    /// and up to and including `split_times[i]`)
+    ///
+    /// Stream n, where n = split_times.len()): Rows that have `time`
+    /// *after* `split_times[n-1]` as well as and NULL rows
```
Comment (Joe-Blount) [inline iox_query/src/frontend/reorg.rs:129]:
```suggestion
    /// *after* `split_times[n-1]` as well as NULL rows
```

File: iox_query/src/frontend/reorg.rs [hunk 2]
```diff
@@ -107,16 +109,31 @@ impl ReorgPlanner {
...
     ///
-    /// The output execution plan has n "output streams" (DataFusion partition):
     /// Stream 0: Rows that have `time` *on or before* the `split_times[0]`
-    /// Stream i (0 < i < split_times's length): Rows that have  `time` in range `(split_times[i-1], split_times[i]]`
-    /// Stream n (n = split_times.len()): Rows that have `time` *after* all the split_times and NULL rows
     ///
-    /// For example, if the input looks like:
+    /// Stream i, where 0 < i < split_times.len(): Rows that have
+    /// `time` in range `(split_times[i-1], split_times[i]]` (greater than split_times[-1]
+    /// and up to and including `split_times[i]`)
```
Comment (Joe-Blount) [inline iox_query/src/frontend/reorg.rs:126]: I had to stare at that a second to realize you're stating the same thing twice.
Separating the repeat on its own line might help.  something like:
```suggestion
    /// Stream i, where 0 < i < split_times.len(): 
    /// Rows have: `time` in range `(split_times[i-1], split_times[i]]`,
    ///  Which is: greater than `split_times[i-1]` up to and including `split_times[i]`.
```

**Review by Joe-Blount — APPROVED** [review]
LGTM.
a couple minor comments, that can be ignored

**Review by NGA-TRAN — APPROVED** [review]
Reviewing this remind me the exact meaning.

File: compactor2/src/plan_ir.rs [hunk 3]
```diff
@@ -20,10 +20,8 @@ pub enum PlanIR {
         /// If there are n split entries in split_times,
         /// there will be `n+1` output files.
         ///
-        /// The contents of each file:
-        /// * `0`: Rows that have `time` *on or before* the `split_times[0]`
-        /// * `i (0 < i < split_times's length)`: Rows that have  `time` in range `(split_times[i-1], split_times[i]]`
-        /// * `n (n = split_times.len())`: Rows that have `time` *after* all the `split_times` and NULL rows
+        /// The distirbution of times is described on
```
Comment (alamb) [inline compactor2/src/plan_ir.rs:23]:
```suggestion
        /// The distribution of times is described on
```

**Pull request merged**